### PR TITLE
Issues/18 - Delegators REST APIs

### PR DESCRIPTION
**Pull request by meywood**

### CODE REVIEW

File: store-api/src/main/java/com/stormeye/event/api/resource/ValidatorResource.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+        logger.debug("getValidatorRewards publicKey {}, page {}, size {}, orderBy {}, orderDirection {}",
+                publicKey,
+                page,
+                size,
+                orderBy,
+                orderDirection
+        );
+
+        var request = PageRequest.of(page - 1, size, getSort(orderBy, orderDirection));
```
Comment (stormeye2000) [inline store-api/src/main/java/com/stormeye/event/api/resource/ValidatorResource.java:90]: PageUtils

File: store-api/src/main/java/com/stormeye/event/api/resource/ValidatorResource.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+     * @return the total rewards
+     * @throws NoSuchAlgorithmException on invalid key
+     */
+    @GetMapping(value = "/validators/{publicKey}/total-rewards", produces = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(tags = "Validators", summary = "Obtains the validator's total rewards",
+            description = "Obtains the validator's total rewards")
+    ResponseEntity<BigInteger> getTotalValidatorRewards(@Parameter(description = "The public key of the validator whose total rewards are to be obtained")
+                                                        @PathVariable(value = "publicKey") final String publicKey) throws NoSuchAlgorithmException {
+
+        logger.debug("getTotalValidatorRewards publicKey {}", publicKey);
```
Comment (stormeye2000) [inline store-api/src/main/java/com/stormeye/event/api/resource/ValidatorResource.java:110]: validate public key

File: store-api/src/main/java/com/stormeye/event/api/resource/ValidatorResource.java [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+        logger.debug("getValidatorBlocks publicKey {}, page {}, size {}, orderBy {}, orderDirection {}",
+                publicKey,
+                page,
+                size,
+                orderBy,
+                orderDirection
+        );
+
+        var request = PageRequest.of(page - 1, size, getSort(orderBy, orderDirection));
```
Comment (stormeye2000) [inline store-api/src/main/java/com/stormeye/event/api/resource/ValidatorResource.java:167]: PageUtils

File: store-api/src/main/java/com/stormeye/event/api/resource/ValidatorResource.java [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+                                                           @RequestParam(value = "page", defaultValue = "1", required = false) final int page,
+                                                           @Parameter(description = "The number of validator rewards to retrieved in a page, defaults to 10")
+                                                           @RequestParam(value = "size", defaultValue = "10", required = false) final int size,
+                                                           @Parameter(description = "The name of the field to sort on")
+                                                           @RequestParam(value = "order_by", defaultValue = TIMESTAMP, required = false) final BlockSortableFields orderBy,
+                                                           @Parameter(description = "The direction of the sort")
+                                                           @RequestParam(value = "order_direction", defaultValue = "DESC", required = false) final Sort.Direction orderDirection) throws NoSuchAlgorithmException {
+
+        logger.debug("getValidatorBlocks publicKey {}, page {}, size {}, orderBy {}, orderDirection {}",
```
Comment (stormeye2000) [inline store-api/src/main/java/com/stormeye/event/api/resource/ValidatorResource.java:159]: Validate Public Key

File: store-api/src/main/java/com/stormeye/event/api/resource/ValidatorResource.java (hunk 1 above)
Comment (meywood) [reply store-api/src/main/java/com/stormeye/event/api/resource/ValidatorResource.java:90]: getSort is in ResourceUtils

Comment (meywood) [reply store-api/src/main/java/com/stormeye/event/api/resource/ValidatorResource.java]: Done

File: store-api/src/main/java/com/stormeye/event/api/resource/ValidatorResource.java (hunk 2 above)
Comment (meywood) [reply store-api/src/main/java/com/stormeye/event/api/resource/ValidatorResource.java:107]: Will be done in https://github.com/stormeye2000/cspr-events-store/issues/23

**Review by stormeye2000 — APPROVED** [review]

**Pull request merged**